pandas/core/categorical.py

```diff
@@ -1995,14 +1995,18 @@ def is_dtype_equal(self, other):
 
         Parameters
         ----------
-        other : Categorical
+        other : Categorical, Series
 
         Returns
         -------
         are_equal : boolean
         """
-
         try:
+            from pandas.core.series import Series
+
+            if isinstance(other, Series):
```

Comment:
the bug is this.

This should take the ordered flag from the passed Categorical/CategoricalIndex
```
In [2]: c = pd.Categorical(list('bac'), ordered=True)

In [3]: c
Out[3]: 
[b, a, c]
Categories (3, object): [a < b < c]

In [4]: pd.Categorical(c)
Out[4]: 
[b, a, c]
Categories (3, object): [a, b, c]
```

so it needs to default to ``ordered=None``. If its not specified it can take the attribute of a passed categorical (we do this already for categories). otherwise, will be ``False``